ReplicatedListExample/ListActor.cs

```diff
@@ -0,0 +1,359 @@
+    {
+        return y.CompareTo(x);
+    }
+}
+
+[MDAutoRegister]
+public class ListActor : Node2D
+{
+    public const string GROUP_ACTORS = "LIST_ACTORS";
+    [MDReplicated]
+    [MDReplicatedSetting(MDReplicatedCommandReplicator.Settings.OnValueChangedEvent, nameof(OnStringListChangedChanged))]
+    [MDReplicatedSetting(MDList.Settings.COMPARATOR, typeof(MDListICompareNormal))] // Valid
+    [MDReplicatedSetting(MDList.Settings.COMPARATOR, typeof(MDListICompareNormal))] // Invalid, already in list
+    [MDReplicatedSetting(MDList.Settings.COMPARATOR, typeof(ListActor))] // Invalid, does not implement ICompare<string>
+    [MDReplicatedSetting(MDList.Settings.COMPARATOR, nameof(MDListICompareNormal))] // Invalid, not a type
+    [MDReplicatedSetting(MDList.Settings.COMPARATOR, typeof(MDListICompareInverse))] // Valid
+    [MDReplicatedSetting(MDList.Settings.UNSAFE_MODE, true)]
+    private MDList<String> ReplicatedStringList;
+
+    [MDBindNode("CanvasLayer/ItemList")]
+    private ItemList DisplayList;
+
+    [MDBindNode("CanvasLayer/Controls/GridContainer/TxtAddItem")]
+    private TextEdit ItemInput;
+
+    [MDBindNode("CanvasLayer/Controls/ChangeNetworkMaster")]
+    private MenuButton PopupMenu;
+
+    private MDGameSession GameSession;
+
+    // Called when the node enters the scene tree for the first time.
+    public override void _Ready()
+    {
+        AddToGroup(GROUP_ACTORS);
+        GameSession = MDStatics.GetGameSession();
+
+        // setup popup
+        if (MDStatics.IsServer())
+        {
+            GameSession.OnPlayerJoinedEvent += OnPlayerJoinedOrLeft;
+            GameSession.OnPlayerLeftEvent += OnPlayerJoinedOrLeft;
+            OnPlayerJoinedOrLeft(0);
+            PopupMenu.GetPopup().Connect("id_pressed", this, nameof(OnChangeNetworkMasterPressed));
+        } 
+        else
+        {
+            PopupMenu.Visible = false;
+        }
+
+        if (IsNotAllowedToModify())
+        {
+            return;
+        }
+        ReplicatedStringList.Add("Initial List String 01");
+        ReplicatedStringList.Add("Duplicate 01");
+        ReplicatedStringList.Add("Initial List String 02");
+        ReplicatedStringList.Add("Duplicate 01");
+        ReplicatedStringList.Add("Duplicate 02");
+        ReplicatedStringList.Add("Duplicate 01");
+        ReplicatedStringList.Add("Intitial List String 03");
+        ReplicatedStringList.Add("Duplicate 02");
+        ReplicatedStringList.Add("Duplicate 01");
+        RefreshList();
+    }
+
+    public override void _ExitTree()
+    {
+        GameSession.OnPlayerJoinedEvent -= OnPlayerJoinedOrLeft;
+        GameSession.OnPlayerLeftEvent -= OnPlayerJoinedOrLeft;
+    }
+
+    protected virtual void OnPlayerJoinedOrLeft(int PeerId)
+    {
+        if (MDStatics.IsNetworkActive() && PeerId == GetNetworkMaster())
+        {
+            // Reset to server if master left
+            this.ChangeNetworkMaster(MDStatics.GetServerId());
```

Comment:
I feel like we should make it automatic in the framework. Maybe have it in a virtual function so users can pick a fallback other than the server

Reply:
(not needed now - just a suggestion)

Reply:
Well I think in normal circumstances you usually clean up nodes owned by someone who left don't you? So if you want to clean it up or if you want to change owner depends on the situation.
Perhaps we can introduce a new concept where any networked node can implement the method MDReplicatedNodeOwnerLeft() and we will look for it when the owner of the node leaves. If we find it we call it, then they can either clean it up or change the owner themselves.